src/Http/Routing/src/RouteOptions.cs

```diff
@@ -152,7 +153,7 @@ public void SetParameterPolicy(string token, [DynamicallyAccessedMembers(Dynamic
         _constraintTypeMap[token] = type;
     }
 
-    private static void AddConstraint<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TConstraint>(Dictionary<string, Type> constraintMap, string text) where TConstraint : IRouteConstraint
+    private static void AddConstraint<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TConstraint>(IDictionary<string, Type> constraintMap, string text) where TConstraint : IRouteConstraint
```

Comment:
```suggestion
    private static void AddConstraint<[DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicConstructors)] TConstraint>(Dictionary<string, Type> constraintMap, string text) where TConstraint : IRouteConstraint
```

This should be reverted. See https://github.com/dotnet/runtime/issues/51193. When we enable that new analyzer we will need to fix it anyway.